fashion-messaging-app-api/server.js

```diff
@@ -37,38 +37,47 @@ app.get('/photos', async (req, res) => {
   
   try {
     const response = await clientAPI.photos.search({ query, per_page: PHOTOS });
-   
-    var myHeaders = new Headers();
-    myHeaders.append("x-api-key", "4d552e9f30522b1dec7c712f83c67c235be86e25ec14b4ba3493383ec7b81d3f");                               
-
-    let fetchPromises = await Promise.all(response.photos.map(async photo => { 
-      var formdata = new FormData();
-      formdata.append("image_url", photo.src.original);
-    
-
-      var requestOptions = {
-        method: 'POST',
-        headers: myHeaders,
-        body: formdata,
-      };
-
-      try {
-        const fetchResponse = await fetch('https://cloudapi.lykdat.com/v1/detection/items', requestOptions);
-        const result = await fetchResponse.json();     
+    let fetchPromises = response.photos.map(async photo => {
+      if (cache.has(photo.src.original)) {
+        return cache.get(photo.src.original);
+      
+      } else {
+        var myHeaders = new Headers();
+        myHeaders.append("x-api-key", "4d552e9f30522b1dec7c712f83c67c235be86e25ec14b4ba3493383ec7b81d3f");                               
 
-        if (result.data.detected_items.length > 0) {
-          return photo;     
+        var formdata = new FormData();
+        formdata.append("image_url", photo.src.original);
+      
+        var requestOptions = {
+          method: 'POST',
+          headers: myHeaders,
+          body: formdata,
+        };
+
+        try {
+          const fetchResponse = await fetch('https://cloudapi.lykdat.com/v1/detection/items', requestOptions);
+          if (fetchResponse.status === 400) {
+            throw new Error('Image size too large for the API.');
+          }
```

Comment:
I got an error after caching my images that did not allow the images to be displayed so I added this and line 71 to skip those photos